src/gui/text/qfontengine.cpp

```diff
@@ -584,10 +584,10 @@ Q_GUI_EXPORT void qt_addBitmapToPath(qreal x0, qreal y0, const uchar *image_data
    // set up edges
    for (int y = 0; y <= h; ++y) {
       for (int x = 0; x <= w; ++x) {
-         bool topLeft = (x == 0) | (y == 0) ? false : SET(x - 1, y - 1);
-         bool topRight = (x == w) | (y == 0) ? false : SET(x, y - 1);
-         bool bottomLeft = (x == 0) | (y == h) ? false : SET(x - 1, y);
-         bool bottomRight = (x == w) | (y == h) ? false : SET(x, y);
+         bool topLeft = ((x == 0) | (y == 0)) ? false : SET(x - 1, y - 1);
+         bool topRight = ((x == w) | (y == 0)) ? false : SET(x, y - 1);
+         bool bottomLeft = ((x == 0) | (y == h)) ? false : SET(x - 1, y);
+         bool bottomRight = ((x == w) | (y == h)) ? false : SET(x, y);
```

Comment:
Doesn't a logical OR (i.e. ||) make more sense here than a bitwise OR (i.e. |)?

Reply:
Perhaps. That could be another good change here. That is a different thing though.

Reply:
Thanks for looking at this, we will check this out.